File: cluster/play-cluster/src/main/scala/play/api/cluster/typed/ClusterBootstrapComponents.scala
/*
 * Copyright (C) Lightbend Inc. <https://www.lightbend.com>
 */

package play.api.cluster.typed

import akka.actor.ActorSystem

import play.api.Configuration
import play.api.Environment
import akka.actor.typed.scaladsl.adapter._
import akka.annotation.ApiMayChange

@ApiMayChange
trait ClusterBootstrapComponents {

  def actorSystem: ActorSystem
  def configuration: Configuration
  def environment: Environment

  // eagerly bootstrap the cluster
  private val _ = new PlayClusterBootstrap(actorSystem, configuration, environment)
}

File: cluster/play-cluster/src/main/java/play/cluster/typed/ClusterBootstrapComponents.java
/*
 * Copyright (C) Lightbend Inc. <https://www.lightbend.com>
 */

package play.cluster.typed;

import play.components.*;
import akka.annotation.ApiMayChange;
import play.api.cluster.typed.PlayClusterBootstrap;
import akka.actor.ActorSystem;
import play.api.Configuration;
import play.api.Environment;

@ApiMayChange
public interface ClusterBootstrapComponents extends AkkaComponents {

  ActorSystem actorSystem();

  Configuration configuration();

  Environment environment();

  default PlayClusterBootstrap playClusterBootstrap() {
    // eagerly bootstrap the cluster
    return new PlayClusterBootstrap(actorSystem(), configuration(), environment());
  }
}
